ideafast/ideafast-patientapp
# Added Devices Screen UIs

Closes: #8

- [x] Assigned a Reviewer to the **PR**.
- [x] Assigned a Project to the **PR**.

## What Changed and Why?

  1. Designed and implemented Devices Screen UI
  1. Added Support for viewing both mock/remote devices in the devices screen

## How to Review Changes?

  1. Review code
  1. View [the prior adobe XD design](
    https://xd.adobe.com/view/d8cb1f48-e69d-4acb-b97d-9ec34c15f0b7-8d98/screen/e0cf22db-0b66-4d52-a9ba-8635ffd3aa77
)
  1. **IMPORTANT**: run `npx react-native link` to generate local fonts for rendering icons
  1. Navigate to the devices screen. You'll now see 'loading' rendering, then the devices' components will be displayed. 
  1. Change the `.env` to use the public/remote API. Rebuild app and visit the devices screen and it'll be the same as the mock API data locally mirrors remote -- key difference is that networked API requests were made. 
  1. Test the pull down to refresh feature.
  1. Test that the translated content loads and renders correctly by changing language to German then Dutch and using the devices screen.
  1. Note: I also added a minor change to the Support screen to [sort the device list by name](https://github.com/ideafast/ideafast-patientapp/pull/60/commits/eab85036ab37e6df04a2960f33df35e5b0f05a39), and refactored it to reuse the Devices component, so worth opening/viewing that screen too.

**Note:** this PR is more about the UI of the devices screen, but since it's so tighly coupled I added an initial logic/implementation (a hook) that we can build upon in a separate PR.
